File: mobile-desktop/mobile_desktop/lib/features/study_sections/writing/models/essay_submission_response.dart
import 'topic_model.dart';

class EssaySubmissionResponse {
  final int id;
  final Topic topic;
  final String content;
  final String gradingType;
  final String? aiFeedback;
  final double? score;

  EssaySubmissionResponse({
    required this.id,
    required this.topic,
    required this.content,
    required this.gradingType,
    this.aiFeedback,
    this.score,
  });

  factory EssaySubmissionResponse.fromJson(Map<String, dynamic> json) {
    return EssaySubmissionResponse(
      id: json['id'],
      topic: Topic.fromJson(json['topic']),
      content: json['content'],
      gradingType: json['gradingType'],
      aiFeedback: json['aiFeedback'],
      score: json['score'] != null ? (json['score'] as num).toDouble() : null,
    );
  }
}

File: backend/backend/src/main/java/com/groupone/backend/dto/EssaySubmissionResponse.java
package com.groupone.backend.dto;

import com.groupone.backend.model.EssaySubmission.GradingType;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class EssaySubmissionResponse {
    private Long id;
    private TopicDto topic;
    private String content;
    private GradingType gradingType;
    private String aiFeedback;
    private Double score;
    private LocalDateTime createdAt;
}
